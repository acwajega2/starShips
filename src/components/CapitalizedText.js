import React, { Component } from 'react';

class CapitalizedText extends Component {
  
    render() {
        const jsUcfirst = (string) => {
            return string.charAt(0).toUpperCase() + string.slice(1);
             
         }
    
        return (
            jsUcfirst(this.props.text)
            
        );
    }
}

export default CapitalizedText;