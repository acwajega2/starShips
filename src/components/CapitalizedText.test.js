import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from '../Enzyme';
import CapitalizedText from './CapitalizedText'





describe("Testing the CapitalizedText Component", () => {
    
//------Doing SnapShot Testing---------------------------
    it("Matches the previous snap shot", () => {
        const sampleText ="hello wajega"
        const component = renderer.create(<CapitalizedText text ={sampleText}/>).toJSON();
        expect(component).toMatchSnapshot()
    });

//------- End of Snapshot Testing--------------------

//----- Testing if the StarShip Provider Initializes Properly
    it("Should Initialize properly", () => {
        const sampleText ="hello wajega"
        shallow(<CapitalizedText text ={sampleText}/>)
    })

    
//---------End of Initialization Testing
    



})