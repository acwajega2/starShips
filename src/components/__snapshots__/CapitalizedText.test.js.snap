exports[`Testing the CapitalizedText Component Matches the previous snap shot 1`] = `"Hello wajega"`;
